src/NServiceBus.NHibernate/Outbox/NHibernateOutboxTransaction.cs

```diff
@@ -25,17 +25,23 @@ public void Dispose()
 
         public void OnSaveChanges(Func<Task> callback)
         {
-            callbacks.Add(callback);
+            if (onSaveChangesCallback != null)
+            {
+                throw new Exception("Save changes callback for this session has already been registered.");
+            }
+            onSaveChangesCallback = callback;
         }
 
         public async Task Commit()
         {
-            await callbacks.InvokeAll().ConfigureAwait(false);
-
+            if (onSaveChangesCallback != null)
```

Comment:
I think you can get rid of this `if` statement using the elvis operator (`?.`) and the `Invoke` method.